File: houston-app/src/main/scala/org/bentocorp/houston/util/PhoneUtils.scala
package org.bentocorp.houston.util

object PhoneUtils {
  def normalize_phone(phone: String, ccIso: String = "US"): String = {
    if(phone.isEmpty) {
      return ""
    }
    var res = phone.replaceAll("\\(|\\)|\\-|\\s", "")
    if ("US".equals(ccIso)) {
      if (res.charAt(0) != '+') {
        if (res.length <= 10) {
          res = "1" + res
        }
        res = "+" + res
      }
    }
    res
  }
}

File: houston-core/src/main/java/org/bentocorp/houston/util/PhoneUtils.java
package org.bentocorp.houston.util;

public class PhoneUtils {

    public enum CountryCode {
        US
    }

    public static String normalize(String phone) {
        return PhoneUtils.normalize(phone, CountryCode.US);
    }

    public static String normalize(String phone, CountryCode cc) {
        String res = phone.replaceAll("\\(|\\)|\\-|\\s", "");
        if (!res.isEmpty() && cc == CountryCode.US) {
            if (res.charAt(0) != '+') {
                if (res.length() <= 10) {
                    res = "1" + res;
                }
                res = "+" + res;
            }
        }
       return res;
    }
}
